modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3235,9 +3259,26 @@ void GDScriptAnalyzer::reduce_preload(GDScriptParser::PreloadNode *p_preload) {
 			}
 		} else {
 			// TODO: Don't load if validating: use completion cache.
-			p_preload->resource = ResourceLoader::load(p_preload->resolved_path);
-			if (p_preload->resource.is_null()) {
-				push_error(vformat(R"(Could not preload resource file "%s".)", p_preload->resolved_path), p_preload->path);
+
+			// Must load GDScript and PackedScenes separately to permit cyclic references
+			// as ResourceLoader::load() detect and reject those.
+			if (ResourceLoader::get_resource_type(p_preload->resolved_path) == "GDScript") {
+				Ref<GDScript> res = GDScriptCache::get_shallow_script(p_preload->resolved_path, parser->script_path);
+				p_preload->resource = res;
+				if (p_preload->resource.is_null()) {
+					push_error(vformat(R"(Could not preload resource script "%s".)", p_preload->resolved_path), p_preload->path);
+				}
+			} else if (ResourceLoader::get_resource_type(p_preload->resolved_path) == "PackedScene") {
```

Comment:
What would happen with other types of resources (maybe custom ones) that may have a reference to a script?

Reply:
We're in `reduce_preload()`, so it parses the `preload()` value. If a GDScript is loaded otherwise, through other resources, the standard loader should work.